tiledb/sm/filesystem/gcs.cc

```diff
@@ -127,20 +130,68 @@ Status GCS::init(const Config& config, ThreadPool* const thread_pool) {
   return Status::Ok();
 }
 
+/**
+ * Builds a chain of service account impersonation credentials.
+ *
+ * @param credentials The set of credentials to start the chain.
+ * @param service_accounts A comma-separated list of service accounts, where
+ * each account will be used to impersonate the next.
+ * @options Options to set to the credentials.
+ * @return The new set of credentials.
+ */
+static shared_ptr<google::cloud::Credentials> apply_impersonation(
```

Comment:
This should definitely have tests checking the edge cases for processing that list of delegates.

Reply:
Added tests for simple and chained impersonation that verify that the impersonation credentials get passed correctly to the SDK. I will speak with some people for manually testing this on the real cloud.